engine/runtime/src/main/scala/org/enso/compiler/PackageRepository.scala

```diff
@@ -0,0 +1,40 @@
+package org.enso.compiler
+
+import org.enso.interpreter.runtime.Context
+import org.enso.interpreter.runtime.builtin.Builtins
+
+import scala.jdk.CollectionConverters._
+
+trait PackageRepository {
+
+  /** Informs the repository that it should populate the top scope with modules
+    * belonging to a given package.
+    *
+    * @param namespace the namespace of the package.
+    * @param name the package name.
+    * @return `true` if the package was already loaded or successfully
+    *         downloaded. `false` otherwise.
+    */
+  def ensurePackageIsLoaded(namespace: String, name: String): Boolean
```

Comment:
I would suggest extending this signature a bit, although I guess it is ok if it happens in future PRs.

I think we should here somehow differentiate various kinds of errors - for display purposes there should be a different error saying "package is not defined in the edition that you are using" versus "package could not be downloaded due to network problems" or "could not load the package due to insufficient filesystem permissions", as these are completely different error, so thay cannot be merged into a single `false` return value.

Reply:
Fair!